statsmodels/genmod/generalized_linear_model.py

```diff
@@ -700,18 +713,82 @@ def estimate_tweedie_power(self, mu, method='brentq', low=1.01, high=5.):
         power : float
             The estimated shape or power
         """
+        if exposure is not None and not isinstance(self.family.link,
+                                                   families.links.Log):
+            raise ValueError("exposure can only be used with the log link function")
+
+        # Use fit exposure if appropriate
+        if exposure is None and hasattr(self, 'exposure'):
+            # Already logged
+            exposure = np.exp(self.exposure)
+        elif exposure is None:
+            exposure = np.ones(len(self.endog))
+
+        # Use fit freq_weights if appropriate
+        if freq_weights is None and hasattr(self, 'freq_weights'):
+            # Already logged
+            freq_weights = self.freq_weights
+        elif exposure is None:
+            freq_weights = np.ones(len(self.endog))
+
+        weights = freq_weights * exposure
```

Comment:
If we don't have reference cases to verify that the code/formulas are correct, then we need to go back to simulating tweedie distributions and check the results, and possibly do a small monte carlo.

I am not able to check the correctness untill I have a large enough block of time to go through formula derivation again (and my notes are in Montreal).

my guess: exposure should automatically downweight observations with small exposure and we shouldn't need to do it explicitly through the weights. I think chi2/dispersion will also need to be adjusted for exposure.

In your 1-day example the downweighting should occur through exposure in our case, and through small var_weights (large variance) in the case where endog is divided by exposure. (unless I have a basic mistake in my thinking.)